ctru-rs/src/services/cam.rs

```diff
@@ -744,6 +745,11 @@ pub trait Camera {
             ))?;
         };
 
+        let screen_size = u32::from(width) * u32::from(height) * 2;
+        if buffer.len() < screen_size as usize {
+            panic!("Provided buffer's length is shorter than the desired width and height matrix.")
```

Comment:
Could it maybe just return an `Err` here instead of panicking (since this function already returns `Result`)?

Reply:
That was my initial approach as well, but the Result returns only our own `Error` type, so I’d need a new enum variant. I’ll see what we can do 👍

Reply:
Made a new error variant with some info.